pkg/virt-controller/services/template.go

```diff
@@ -73,6 +73,39 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VirtualMachine) (*kubev1.P
 	// the qemu pid is in the pods mount namespace
 	privateDir := fmt.Sprintf("%s-private/%s/%s", t.virtShareDir, namespace, domain)
 
+	// Consider CPU and memory requests and limits for pod scheduling
+	resources := kubev1.ResourceRequirements{}
+	vmResources := vm.Spec.Domain.Resources
+	requests := kubev1.ResourceList{}
```

Comment:
I think that you can just directly attach requests and limits to the resources. No need for the temporary maps. It should not map for the serializer if they are empty or nil.